syft/pkg/cataloger/golang/parse_go_binary.go

```diff
@@ -37,15 +38,16 @@ var (
 	// inject the correct version into the main module of the build process
 
 	knownBuildFlagPatterns = []*regexp.Regexp{
-		regexp.MustCompile(`(?m)\.([gG]it)?([bB]uild)?[vV]ersion=(\S+/)*(?P<version>v?\d+.\d+.\d+[-\w]*)`),
+		regexp.MustCompile(`(?m)\.([gG]it)?([bB]uild)?[vV]er(sion)?=(\S+/)*(?P<version>v?\d+.\d+.\d+[-\w]*)`),
```

Comment:
adjusts the pattern to be more permissive based on examples in the field. This allows for us to depend on the (probably more dependable) LD flags instead of the binary contents pattern matching in more cases.